UI: New text-strings preview mode

KWMORALE opened the pull request:
<!-- Thank you for contributing! -->
<!-- Please fill this template before submitting your PR (fill the boxes using "x") -->

**Your checklist for this pull request**
- [ ] I've read the [contributing guideline](CONTRIBUTING.md).
- [ ] I've tested my changes by building and running the project, and testing changed functionality (if applicable)
- [ ] I've added automated tests for my change (if applicable, optional)
- [ ] I've updated documentation to reflect my change (if applicable)

**What is the current behaviour?**
<!-- Explain how the code works currently -->
Only raw and hex mode in sample preview.

**What is the new behaviour?**
<!-- Explain how the code works after your changes -->
- Added a dropdown to select the file content display mode.
- Rename component `HexView` to `ObjectPreview`
- New preview mode: text - that prints the printable character sequences that are at least 

**Test plan**
<!-- Explain how to test your changes -->

<!-- After submitting, your code will be tested by the CI pipeline. Please
ensure that all tests pass. If they don't at first, please update your code -->
Go to sample preview and change mode on some files.

[Screencast from 2025-07-23 12-07-29.webm](https://github.com/user-attachments/assets/47e4086d-59e0-4d5c-8eec-57662b54fa24)

postrowinski commented on mwdb/web/src/commons/ui/ObjectPreview.tsx:
```diff
@@ -76,13 +76,39 @@ export function HexView(props: Props) {
             if (props.content instanceof ArrayBuffer)
                 return new TextDecoder().decode(props.content);
             else return props.content;
+        } else if (props.mode === "text") {
```
Currently, there are too many if-logic blocks here. I suggest creating separate utility functions for each mode — 'raw', 'text', and 'hex' to handle the value for each of them individually.

KWMORALE replied:
Refactored the code and added widechar utf-16le support. Please test this really extensively.

postrowinski commented on mwdb/web/src/commons/ui/ObjectPreview.tsx:
```diff
@@ -76,13 +76,39 @@ export function HexView(props: Props) {
+            else buffer = new TextEncoder().encode(props.content);
+
+            const MIN_LENGTH = 4;
+            let result: string[] = [];
```
```suggestion
           const result: string[] = [];
```

postrowinski commented on mwdb/web/src/components/PreviewSwitchAction.tsx:
```diff
@@ -1,21 +1,40 @@
+            elements={modes.map((mode) => {
+                const selected = currentMode === mode;
+
+                const selectedStyle = {
+                    fontWeight: selected ? "bold" : "normal",
+                    pointerEvents: (selected ? "none" : "auto") as
+                        | "none"
+                        | "auto",
+                };
```
```suggestion
const styles = { fontWeight: selected ? "bold" : "normal" };
```

postrowinski commented on mwdb/web/src/components/PreviewSwitchAction.tsx:
```diff
@@ -1,21 +1,40 @@
+                };
+
+                return (
+                    <span className="nav-item" style={selectedStyle} key={mode}>
```
```suggestion
                    <span className="nav-item" style={styles} key={mode}>
```

psrok1 replied:
or `navItemStyle`

KWMORALE commented:
Conditions have been moved to functions and the formatting of the returned value has been moved to individual ones.

Added support for UTF-16(le) as Widechar. Renamed the Text mode to Strings to target Linux UTF-8 characters.
If no characters in the specified format are found, the following message is displayed: "No {format} characters..."

KWMORALE commented:
<img width="1095" height="1090" alt="Screenshot from 2025-07-30 13-08-21" src="https://github.com/user-attachments/assets/46bad81f-3946-48e6-97a9-e3e5106f699b" />
<img width="1095" height="1090" alt="Screenshot from 2025-07-30 13-08-31" src="https://github.com/user-attachments/assets/36795f80-78f9-4e47-8611-51d0c188c940" />

psrok1 commented on mwdb/web/src/commons/helpers/preview.ts:
```diff
@@ -0,0 +1,91 @@
+    if (format === "UTF-16(LE)") {
+        const buffer = getArrayBuffer(content);
+        const totalUnits = Math.floor(buffer.byteLength / 2);
+        const units16 = new Uint16Array(buffer, 0, totalUnits);
```
I think that `chars` would be a better word for these variables

psrok1 reviewed (commented):
Overall OK, but that "No {format} characters" would be very misleading. I would also rename `units` to `chars` - we're not really parsing UTF-8/16 (and we shouldn't) so they're not really characters, but it's much closer description for that value than something abstract.
